node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -109,7 +110,7 @@ object Main {
     implicit val replServiceClient: GrpcReplClient[F] =
       new GrpcReplClient[F](
         options.grpcHost(),
-        options.grpcPort(),
+        replPort(options),
```

Comment:
Sorry for misled you on the path to get `api-server.port-grpc-internal`. Now I see it makes more sense to just change default port number in `grpc-port` config used for repl client.
In the future we will unify configuration with Scala types which will contain defaults and parsing libs will provide overrides. Now _hocon_ configuration (default.conf) serves as the main source which is not what we want.

https://github.com/rchain/rchain/blob/8ba22c07f1166f2d94ecc910c3f56ca686d964e2/node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala#L128-L132

Reply:
@tgrospic, I changed default value for grpcPort to `40402`and reverted previous irrelevant commits.

Reply:
I don't think we want to change `grpcPort` default. Its a public API port, so when binary is in client mode, e.g. `rnode deploy` it should work.
I'm not sure why previous changes were not OK? The goal was to use `rnode eval` internal port by default, and this is what we want, right? Because eval server is running on internal port.

Reply:
@nzpr Yes, I notice that and corrected my comment. We need separate parameter for repl client.

Reply:
For example, this is not helpful. Where am I connecting?
```
$ rnode eval --help
  -p, --print-unmatched-sends-only
  -h, --help                         Show help message

 trailing arguments:
  file-names (required)
```

Reply:
@nzpr, @tgrospic, [resolved](https://github.com/rchain/rchain/pull/3564/commits/0a24a8d8ceaa63d4c53e672581e470715fb73d78)

Reply:
@tgrospic I'm not sure I get this. Rnode binary in server mode (rnode run) has two API servers - external and internal (accessing restricted resources), one 40401, another 40402. Eval API endpoint is available only on internal port.
If you use rnode in client mode, you have --grpc-port and --grpc-host params in action. Then you can choose what you want client call you want to make. Eval is one of client calls.

So `rnode --grpc-port 40402 eval` calls eval on localhost:40402.

Reply:
It's confusing that you need to specify default port to call eval and repl.

Reply:
@tgrospic hm.. I don't get why initial version was not ok? Why not to use default internal port and not `--grpc-port` for eval and repl client calls without adding one more param? Why to add param for this?

Reply:
@nzpr I agree that the new parameter is not the real solution. Port and host are the main problem here which are not sub-commands of grpc client calls so it's confusing for which commands they are used.

You are right, initial solution makes more sense. It's only changing behavior when port is not specified which can hardly break any existing code.

We need to move definition of config with defaults in Scala anyway, so for now it's much easier to just define default for `run.apiPortGrpcInternal` option and use it as alternative value instead of reading default.conf.
```scala
options.grpcPort.orElse(options.run.apiPortGrpcInternal.toOption)()
```
And this also applies to ProposeService client.

Reply:
Do you mean it is more beneficial to have `--grpc-port --grpc-host` under `deploy`, `eval`, `propose`, `repl` - all of these subcommands?
So not make them rnode param, but param of subcommand? 
I'm ok with that, but imo not its fine as well, rnode by default can be treated as a client, so you tell which endpoint to use. And then you specify the API you want to call. Or you can start the rnode server  by `rnode run`

Reply:
> Do you mean it is more beneficial to have --grpc-port --grpc-host under deploy, eval, propose, repl - all of these subcommands?

Yes.

> rnode binary by default can be treated as a client

RNode is a client but with deploy, eval, ... commands its purpose is to relay messages to running client. And why would gRPC protocol be something special.
Something like this `rnode eval --http --port 443 deploy ...`. 

---

We should see configuration as as small VM which can have many interfaces which all can work in a similar way.
Each time command is called user must wait on initial loading of JVM which makes the whole command look slow. So for that we can create interactive console using the same API in the background.

```
$ rnode console
> RNode version v0.12.4 relay console
> set protocol http
> OK
> set external-port 443
> OK
> status
> {
  "address":"rnode://4a9...67f6@node2.testnet.rchain.coop?protocol=42400&discovery=42404",
  "version":"RChain Node 0.12.4 (f9ebcbd93ef308ca1175ccff7aeaca8a81da34cc)",
  "peers":7,
  "nodes":34
}
> set private-key "0x1234..."
> OK
> deploy --phlo-limit 200000 ...
> OK signature 1234...
```